components/utilities/dialog/index.jsx

```diff
@@ -412,15 +427,18 @@ const Dialog = createReactClass({
 							[`${this.props.outsideClickIgnoreClass}`]:
 								this.props.position === 'overflowBoundaryElement',
 						},
-						this.props.contentsClassName
+						this.props.contentsClassName,
+						this.props.hasNubbin && getNubbinClassName(this.props.align),
 					) || undefined
 				}
+				id={this.props.id}
 				style={style}
 				onKeyDown={this.handleKeyDown}
 				onMouseEnter={this.props.onMouseEnter}
 				onMouseLeave={this.props.onMouseLeave}
 				{...this.props.containerProps}
 				ref={this.setDialogContent}
+				role={this.props.variant}
```

Comment:
Not sure how to address this properly.

Reply:
This is fine.

Reply:
Yeah that's what I was thinking, but that's only if the direct variant -> role mapping ends up insufficient.